rumqttc/src/v5/state.rs

```diff
@@ -64,6 +64,8 @@ pub enum StateError {
     PubCompFail { reason: PubCompReason },
     #[error("Connection failed with reason '{reason:?}' ")]
     ConnFail { reason: ConnectReturnCode },
+    #[error("Connection closed by peer abruptly")]
+    ConnectionAborted
```

Comment:
Would love to have this named `ConnectionClosed`. Think this is the term use widely when a `read` returns `0`.